packages/core/src/particle/modules/MainModule.ts

```
@@ -293,8 +293,7 @@ export class MainModule implements ICustomClone {
       case ParticleSimulationSpace.Local:
         shaderData.setVector3(MainModule._worldPosition, transform.worldPosition);
         const worldRotation = transform.worldRotationQuaternion;
-        const worldRotationV4 = MainModule._tempVector40;
-        worldRotationV4.copyFrom(worldRotation);
+        const worldRotationV4 = this._tempVector40.copyFrom(worldRotation); // Maybe shaderData should support Quaternion
```

Biome

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)